packages/web/src/components/notification/Notifications/Notification.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+
+import { useSelector } from 'react-redux'
+
+import { CommonState } from 'common/store'
+import {
+  getNotificationEntities,
+  getNotificationEntity,
+  getNotificationUser,
+  getNotificationUsers
+} from 'common/store/notifications/selectors'
+import {
+  Notification as Notifications,
+  NotificationType
+} from 'common/store/notifications/types'
+
+import { ChallengeRewardNotification } from './ChallengeRewardNotification'
+import { FavoriteNotification } from './FavoriteNotification'
+import { FollowNotification } from './FollowNotification'
+import { MilestoneNotification } from './MilestoneNotification'
+import { RemixCosignNotification } from './RemixCosignNotification'
+import { RemixCreateNotification } from './RemixCreateNotification'
+import { RepostNotification } from './RepostNotification'
+import { TierChangeNotification } from './TierChangeNotification'
+import { TipReactionNotification } from './TipReactionNotification'
+import { TipSentNotification } from './TipSentNotification'
+import { TopSupporterNotification } from './TopSupporterNotification'
+import { TopSupportingNotification } from './TopSupportingNotification'
+import { TrendingTrackNotification } from './TrendingTrackNotification'
+import { UserSubscriptionNotification } from './UserSubscriptionNotification'
+import { USER_LENGTH_LIMIT } from './utils'
+
+type NotificationProps = {
+  notification: Notifications
+}
+
+export const Notification = (props: NotificationProps) => {
+  const { notification: notificationProp } = props
+  const user = useSelector((state: CommonState) =>
+    getNotificationUser(state, notificationProp)
+  )
+
+  const users = useSelector((state: CommonState) =>
+    getNotificationUsers(state, notificationProp, USER_LENGTH_LIMIT)
+  )
+
+  const entity = useSelector((state: CommonState) =>
+    getNotificationEntity(state, notificationProp)
+  )
+
+  const entities = useSelector((state: CommonState) =>
+    getNotificationEntities(state, notificationProp)
+  )
+
+  // Based on how notification types are defined, we need to cast like this.
+  // In the future we should select user/users/entity/entities in each notif.
```

Comment:
+1. I think ideally the NotificationPanel can use some mapping of NotificationType => Notification element and avoid the wrapper in the future

Reply:
yeah the mapping def makes sense! im not sure if the wrapping is smart enough to figure out that if the type is x, then the component is y, and the prop shape is z, though, if that makes sense?

Reply:
I'm thinking in the future these notifs will all have the same props structure so we just pass it in the same to each one, but maybe not?

Reply:
they will be diff in the sense that they have BaseNotif, but then other things like value vs reaction vs whatevs i think!

Reply:
Hmm I'm thinking like:

NotificationPanel.tsx:
```jsx
// ....
       notifications.map(notification =>
           <notificationTypeMap[notification.type] key={notification.id} {...notification} />
```

Is that what you're thinking?

Reply:
yeah the main issue with that with ts though is that `Notification` is a discriminated union, so we need to get the component from the type, and also merge in the notification with the corresponding type, that's why the switch case works here cause it aligns the notification type so the component + notification type checks correctly

Reply:
Makes sense, gotta appease the typescript engine. Also makes it easier to reason about anyway probably. I'm ok with switch cases, the only slight concern for me is the intermediate React component. I feel like I remember that causing wrapping <divs> and/or having perf implications, but it's been a while and I'm probably wrong and it's not an issue anymore or never was...

Reply:
yeah i def hear you on the extra component... will be thinking of a way to make typescript happy while also avoiding a wrapper